frontend/src/domain/Die.ts

```diff
@@ -1,3 +1,7 @@
-export const ALL_DICE = ["d4", "d6", "d8", "d10", "d12", "d20"];
+export const ALL_DICE = ["d4", "d6", "d8", "d10", "d12", "d20"] as const;
 
-export type Die = typeof ALL_DICE[number];
+type ElementType<T extends ReadonlyArray<unknown>> = T extends ReadonlyArray<infer ElementType>
```

Comment:
wat

Reply:
is this saying... like, "if it's in the array of type T, it's a valid ElementType, else it's not a valid ElementType"
?

Reply:
yeah! that's my understanding at least.